### PR TITLE
Add new extract function for detecting image URLs

### PR DESCRIPTION
**Pull request by cdpierse**
- Adds new feature discussed in #1117 and in slack
- very slight changes made to the proposed regex in the above issue

### CODE REVIEW

File: unstructured/nlp/patterns.py [hunk 1]
```diff
@@ -138,3 +138,8 @@
 
 # taken from https://stackoverflow.com/a/3845829/12406158
 VALID_JSON_CHARACTERS = r"[,:{}\[\]0-9.\-+Eaeflnr-u \n\r\t]"
+
+IMAGE_URL_PATTERN = (
+    r"(https?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|"
+    r"(?:%[0-9a-fA-F][0-9a-fA-F]))+\.(?:jpg|jpeg|png|gif|bmp))"
+)
```
Comment (cragwolfe) [inline unstructured/nlp/patterns.py:145]: should be a case-insenstive regex? (to be fair, may apply to other patterns as well)

File: unstructured/nlp/patterns.py [hunk 2]
```diff
@@ -138,3 +138,5 @@
 
 # taken from https://stackoverflow.com/a/3845829/12406158
 VALID_JSON_CHARACTERS = r"[,:{}\[\]0-9.\-+Eaeflnr-u \n\r\t]"
+
+IMAGE_URL_PATTERN = r"(?i)https?://" r"(?:[a-z0-9$-_@.&+!*\\(\\),])+" r"\.(?:jpg|jpeg|png|gif|bmp)"
```
[the remark this answers is not in the document]
Comment (cdpierse) [reply unstructured/nlp/patterns.py:142]: Not sure exactly which part of the pattern it doesn't like here.

**Review by cragwolfe — APPROVED** [review]
LGTM!

**Comment by cdpierse** [comment]
@cragwolfe so it looks like I'm unable to find a regex pattern that satisfies all my tests and does _not_ raise a medium-level CodeQL warning.

I've reverted back to what I had earlier today where there was only 1 warning. I see some other patterns have `# noqa W605` to silence potential false positives with CodeQL, would this be a candidate for that ?

**Comment by cragwolfe** [comment]
@cdpierse , i think this is working:

https://github.com/Unstructured-IO/unstructured/pull/1244/commits/776763607422e5f88af2312e1952d6e37e0380bb

if you want to cherry-pick that, though still needs a `make tidy`

**Comment by cdpierse** [comment]
Thanks @cragwolfe have applied that now, looks like we are all good with both tests and codeQL

**Pull request merged**